src/DynamoCore/Models/HomeWorkspaceModel.cs

```diff
@@ -391,7 +439,20 @@ private void OnUpdateGraphCompleted(AsyncTask task)
 
             OnEvaluationCompleted(e);
 
+            if (EngineController.IsDisposed) return;
+
             EngineController.ReconcileTraceDataAndNotify();
+
+            // Refresh values of nodes that took part in update.
+            foreach (var modifiedNode in updateTask.ModifiedNodes)
+            {
+                modifiedNode.RequestValueUpdateAsync(scheduler, EngineController);
+            }
+
+            scheduler.Tasks.AllComplete(_ =>
+            {
+                OnRefreshCompleted(e);
+            });
```

Comment:
This is the major change in this PR.  Now `PulseMaker` must await all tasks, including refresh to be completed.